Week3/db-week-3.js

```diff
@@ -0,0 +1,216 @@
+'use strict';
+const util = require('util');
+const fs = require('fs');
+const mysql = require('mysql');
+const express = require('express');
+const app = express();
+const bodyParser = require('body-parser');
+const port = 3020;
+
+const todoDatabase = {
+  host: 'localhost',
+  user: 'hyfuser',
+  password: 'hyfpassword',
+  database: 'todo',
+  multipleStatements: true,
+};
+
+app.set('view engine', 'pug');
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(express.static(__dirname + '/public'));
+
+const CREATE_TABLE_TODOS = `CREATE TABLE IF NOT EXISTS todos ( 
+  todoid VARCHAR(30) PRIMARY KEY,
+  reminder text 
+  );`;
+
+const CREATE_TABLE_ITEMS = `CREATE TABLE IF NOT EXISTS items (
+    item VARCHAR(30),
+    isdone enum('Yes', 'No'),
+    belongs_to VARCHAR(30),
+    
+    FOREIGN KEY (belongs_to)
+    REFERENCES todos(todoid)
+    ON DELETE CASCADE
+    );`;
+
+app.get('/', async function(req, res) {
+  try {
+    const connection = mysql.createConnection(todoDatabase);
+    const execQuery = util.promisify(connection.query.bind(connection));
+    await execQuery(CREATE_TABLE_TODOS);
+    await execQuery(CREATE_TABLE_ITEMS);
+    res.sendFile(__dirname + '/pages/index.html');
+    connection.end();
+  } catch (err) {
+    if (err) {
+      throw err;
+    }
+  }
+});
+
+app.get('/download', function(req, res) {
+  const file = `${__dirname}/instructions/Readme.md`;
+  res.download(file);
+});
+
+app.post('/submit', async function(req, res) {
+  try {
+    const connection = mysql.createConnection(todoDatabase);
+    const execQuery = util.promisify(connection.query.bind(connection));
+    const todoQuery = `INSERT INTO todos VALUES ('${req.body.todoid}', '${req.body.reminder}')`;
+    const newList = await execQuery(todoQuery);
+    if (newList.length === 0) {
+      res.render('index', {
+        message: 'A list name or a reminder cannot be empty. Please go back and fill the inputs',
+      });
+    } else {
+      res.render('index', { message: 'Thank you for your patience! Your list is saved!' });
+      connection.end();
+    }
+  } catch (err) {
+    if (err) {
+      throw err;
+    }
+  }
+});
+
+app.get('/addItems', async function(req, res) {
+  try {
+    res.sendFile(__dirname + '/pages/addItems.html');
+  } catch (err) {
+    if (err) {
+      throw err;
+    }
+  }
+});
+
+app.post('/saveAddItem', async (req, res) => {
+  try {
+    const connection = mysql.createConnection(todoDatabase);
+    const execQuery = util.promisify(connection.query.bind(connection));
+    const itemQuery = `INSERT INTO items VALUES ('${req.body.task}', '${req.body.isdone}', '${req.body.listName}')`;

```

Comment:
Just one more option is to use it like this:

```
const task = ${req.body.task};
const isDone = ${req.body.isdone};
const listName = ${req.body.listName};

const itemQuery = `INSERT INTO items VALUES (?, ?, ?)`;
await execQuery(itemQuery, [task, isDone, listName]);
```

Simply because this way:

- sentences are shorter
- more readable for other developers